src/lib/y2storage/guided_proposal.rb

```diff
@@ -120,12 +120,12 @@ def calculate_proposal
       exception = nil
       saved_root_device = populated_settings.root_device
 
-      [:desired, :min].each do |target|
-
+      target_sizes.product(candidate_devices).each do |target, devices|
```

Comment:
NP: it is a bit weird to me that `candidate_devices` actually is a list of groups of candidate devices. I would try to do it more explicit to avoid misleading.

```suggestion
      groups = group_candidate_devices
      target_sizes.product(groups).each do |target, devices|
```

And then,

```
      def group_candidate_devices
        candidate_devices.zip.append(candidate_devices).uniq
      end
```

So, `candidate_devices` does not need to group nothing by itself, it is only the plain list of candidate devices.

Reply:
This looks to me a good improvement. Believe or not, I could not find this more accurate name.